modules/product/src/main/java/com/opengamma/strata/product/index/OvernightFutureOptionPosition.java

```diff
@@ -0,0 +1,635 @@
+import org.joda.beans.gen.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.Resolvable;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.PortfolioItemInfo;
+import com.opengamma.strata.product.PortfolioItemSummary;
+import com.opengamma.strata.product.PositionInfo;
+import com.opengamma.strata.product.ProductType;
+import com.opengamma.strata.product.SecuritizedProductPosition;
+import com.opengamma.strata.product.SecurityId;
+import com.opengamma.strata.product.common.SummarizerUtils;
+
+/**
+ * A position in an option on a futures contract based on an Overnight index.
+ * <p>
+ * A position in an underlying {@link OvernightFutureOption}.
+ * <p>
+ * The net quantity of the position is stored using two fields - {@code longQuantity} and {@code shortQuantity}.
+ * These two fields must not be negative.
+ * In many cases, only a long quantity or short quantity will be present with the other set to zero.
+ * However it is also possible for both to be non-zero, allowing long and short positions to be treated separately.
+ * The net quantity is available via {@link #getQuantity()}.
+ * <p>
+ * Strata uses <i>decimal prices</i> for overnight future options in the trade model, pricers and market data.
+ * The decimal price is based on the decimal rate equivalent to the percentage.
+ * For example, an option price of 0.2 is related to a futures price of 99.32 that implies an
+ * interest rate of 0.68%. Strata represents the price of the future as 0.9932 and thus
+ * represents the price of the option as 0.002.
+ */
+@BeanDefinition(constructorScope = "package")
+public final class OvernightFutureOptionPosition
+    implements
+    SecuritizedProductPosition<OvernightFutureOption>, Resolvable<ResolvedOvernightFutureOptionTrade>, ImmutableBean, Serializable {
+
+  /**
+   * The additional position information, defaulted to an empty instance.
+   * <p>
+   * This allows additional information to be attached to the position.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final PositionInfo info;
+  /**
+   * The option that was traded.
+   * <p>
+   * The product captures the contracted financial details.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final OvernightFutureOption product;
+  /**
+   * The long quantity of the security.
+   * <p>
+   * This is the quantity of the underlying security that is held.
+   * The quantity cannot be negative, as that would imply short selling.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegative")
+  private final double longQuantity;
```

Comment:
How about creating a field "double quantity" and an enum field with two possible values "SHORT", "LONG" ? Maybe it will be more convenient to use?

Reply:
I do not have a strong opinion here, maybe this is the convention which we do use here (creating two separate fields: longQuantity and shortQuantity)

Reply:
Yes, this is the convention used for all the "Position" objects. In some circumstances, we receive both a long and short quantity.